packages/menu/src/Menu.spec.tsx

```diff
@@ -65,94 +84,207 @@ describe('packages/menu', () => {
-        userEvent.type(menu, '{esc}');
-      } else {
+    const userEventInteraction = (el: HTMLElement, key: Keys) => {
+      if (key === 'tab') {
         userEvent.tab();
+      } else {
+        userEvent.type(el, `{${key}}`);
       }
     };
 
-    describe.each(tests)('%s key', key => {
+    describe.each(closeKeys)('%s key', key => {
       test('Closes menu', async () => {
-        const { getByRole, getByTestId } = renderMenu({
-          trigger,
-        });
-        const button = getByRole('button');
-        userEvent.click(button);
+        const { getByTestId, openMenu } = renderMenu();
+        openMenu();
         const menu = getByTestId(menuTestId);
 
         userEventInteraction(menu, key);
         await waitForElementToBeRemoved(menu);
         expect(menu).not.toBeInTheDocument();
       });
+
       test('Returns focus to trigger {usePortal: true}', async () => {
-        const { getByRole, getByTestId } = renderMenu({
-          trigger,
+        const { getByTestId, openMenu, triggerButton } = renderMenu({
           usePortal: true,
         });
-        const button = getByRole('button');
-        userEvent.click(button);
+        openMenu();
         const menu = getByTestId(menuTestId);
 
         userEventInteraction(menu, key);
         await waitForElementToBeRemoved(menu);
-        expect(button).toHaveFocus();
+        expect(triggerButton).toHaveFocus();
       });
 
       test('Returns focus to trigger {usePortal: false}', async () => {
-        const { getByRole, getByTestId } = renderMenu({
-          trigger,
+        const { getByTestId, openMenu, triggerButton } = renderMenu({
           usePortal: false,
         });
-        const button = getByRole('button');
-        userEvent.click(button);
+        openMenu();
         const menu = getByTestId(menuTestId);
 
         userEventInteraction(menu, key);
         await waitForElementToBeRemoved(menu);
-        expect(button).toHaveFocus();
+        expect(triggerButton).toHaveFocus();
+      });
+    });
+
+    describe('Enter key', () => {
+      test('Fires the click handler of the highlighted item', () => {
+        const { getByTestId, openMenu } = renderMenu();
+        openMenu();
+        const menu = getByTestId(menuTestId);
+        const options = globalGetAllByRole(menu, 'menuitem');
+        expect(options[0]).toHaveFocus();
+
+        userEvent.type(options[0], `{enter}`);
+        expect(onClick).toHaveBeenCalled();
+      });
+
+      test('Closes the menu', async () => {
+        const { getByTestId, openMenu, triggerButton } = renderMenu();
+        openMenu();
+        const menu = getByTestId(menuTestId);
+        const options = globalGetAllByRole(menu, 'menuitem');
+
+        userEvent.type(options[0], `{enter}`);
+        expect(triggerButton).toHaveFocus();
+
+        // FIXME: I can verify that this works in the browser,
+        // but this does not work in jest - AT
+        // await waitForElementToBeRemoved(menu);
```

Comment:
I couldn't get these tests to pass even though it works in the browser, and the mouse interaction version of this test is passing (L143)

Reply:
this seems to work `userEvent.click(options[0]);`